includes/actions/update-next-payment.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace AutomateWoo_Subscriptions;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Change a subscription's next payment date.
+ *
+ *
+ * @class Action_Subscription_Update_Next_Payment
+ * @since 1.2.2
+ */
+class Action_Subscription_Update_Next_Payment extends Abstract_Action_Subscription {
+
+	/**
+	 * The next payment date of the subscription before running the action.
+	 *
+	 * @var string
+	 */
+	private $original_payment_date;
+
+	/**
+	 * Explain to store admin what this action does via a unique title and description.
+	 */
+	public function load_admin_details() {
+		parent::load_admin_details();
+		$this->title       = __( 'Update Next Payment', 'automatewoo-subscriptions' );
+		$this->description = __( 'Change a subscription\'s next payment date.', 'automatewoo-subscriptions' );
+	}
+
+	/**
+	 * Edit the item managed by this class on the subscription passed in the workflow's trigger
+	 *
+	 * @throws \Exception When there is an error.
+	 */
+	public function edit_subscription( $subscription ) {
+		$this->original_payment_date = $subscription->get_date( 'next_payment' );
+		$new_payment_date            = $this->get_new_payment_date();
+		$subscription->update_dates( array( 'next_payment' => wcs_get_datetime_utc_string( $new_payment_date ) ) );
+	}
+
+	function load_fields() {
+		$this->load_subscription_renewal_fields();
+	}
```

Comment:
Is there any reason to have `load_subscription_renewal_fields()` has a separate method here? Seems like we could just add the fields directly in `load_fields()`

Reply:
Makes sense. Fixed via [`b1a0cd1`](https://github.com/woocommerce/automatewoo-subscriptions/pull/59/commits/b1a0cd1fdd453712bbbab10895e009e6096d05f1).